stl/inc/concepts

```diff
@@ -108,9 +109,12 @@ concept constructible_from = destructible<_Ty>
     && __is_constructible(_Ty, _ArgTys...);
 
 // CONCEPT default_initializable
-template <class _Ty> // Per P1754R1 and LWG-3149
+template <class _Ty>
 concept default_initializable = constructible_from<_Ty>
-    && requires { _Ty{}; }; // && __is_default_initializable(_Ty); // TRANSITION, VSO-896348
+    && requires {
+        _Ty{};
+        ::new ((void*)nullptr) _Ty; // is-default-initializable<_Ty>

```

Comment:
Wait, there are types which pass the first expression but not this one?!

:sigh:

Reply:
Default-init of `const` objects is ill-formed unless those objects have class type and user-provided default constructors (roughly). So e.g. `new const int[42]` is ill-formed but `const int x[42]{}` is not.

Reply:
I see. Ow

Reply:
I saw this in libstdc++ implementation and  decided to go for a walk rather than trying to implement it for libc++